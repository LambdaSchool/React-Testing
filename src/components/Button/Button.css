.component-button {
    display: inline-flex;
    width: 25%;
    flex: 1 0 auto;
    height: 75px;
    margin: 1px 0;
}

.component-button.wide {
    width: 50%;
}

.component-button button {
    background-color: #E0E0E0;
    border: 0;
    font-size: 30px;
    margin: 0 1px 0 0;
    flex: 1 0 auto;
    padding: 0;
}

.component-button button:hover {
    background-color: rgb(184, 184, 184);
    cursor: pointer;
}

.component-button:last-child button {
    margin-right: 0;
}

.component-button.orange button {
    background-color: #F5923E;
    color: white;
}

.component-button.orange button:hover {
    background-color: rgb(209, 123, 49);
    cursor: pointer;
}