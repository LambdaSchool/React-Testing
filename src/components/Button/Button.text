Button 

props =>  orange  should be a boolean
... wide should be a boolean
... name shoul be a a string

function => handleClick accepts the name as input 




