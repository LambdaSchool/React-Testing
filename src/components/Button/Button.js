import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Button.css';

class Button extends Component {
    handleClick = () => {
        this.props.clickHandler(this.props.name);
    }

    render() {
        const classNames = [
            'component-button',
            this.props.orange ? 'orange' : '',
            this.props.wide ? 'wide' : '',
        ];
        return (
            <div className={classNames.join(' ').trim()}>
                <button onClick={this.handleClick}>
                    {this.props.name}
                </button>
            </div>
        );
    }
}
/* 
- renders the button with the appropriate classNames depedning on what props are passed
- should change the state for next if a number button has been pressed
*/
Button.propTypes = {
    name: PropTypes.string,
    orange: PropTypes.bool,
    wide: PropTypes.bool,
    clickHandler: PropTypes.func,
};

export default Button;