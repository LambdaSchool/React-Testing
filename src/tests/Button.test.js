import React from 'react';
import ReactDOM from 'react-dom';
import sinon from 'sinon';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from '../components/Button/Button';

Enzyme.configure({ adapter: new Adapter() });

describe('<Button />', () => {
  it('Should return correct button name ', () => {
    const wrapper = shallow(
    <Button name={"AC"} orange={true} wide={true} clickHandler={() => {}}/>
  );
  expect(wrapper).toMatchSnapshot()
  });

  it("Button renders correctly", () => {
    const wrapper = render(
      <Button name={"AC"} orange={true} wide={true} clickHandler={() => {}}/>
    );
    expect(wrapper).toMatchSnapshot()
  })

  it("should contain a class of orange wide", () => {
    const wrapper = shallow(<Button orange={true} />);
    expect(wrapper.find("div").first().hasClass("orange")).toEqual(true);
  });
});