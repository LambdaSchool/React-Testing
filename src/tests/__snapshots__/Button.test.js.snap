// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Button renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "component-button orange wide",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "AC",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`<Button /> Should return correct button name  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    clickHandler={[Function]}
    name="AC"
    orange={true}
    wide={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        onClick={[Function]}
>
        AC
</button>,
      "className": "component-button orange wide",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "AC",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "AC",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          onClick={[Function]}
>
          AC
</button>,
        "className": "component-button orange wide",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "AC",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "AC",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
